src/vs/workbench/contrib/files/browser/views/explorerViewer.ts

```diff
@@ -608,8 +607,8 @@ export class FilesFilter implements ITreeFilter<ExplorerItem, FuzzyScore> {
 	private editorsAffectingFilter = new Set<EditorInput>();
 	private _onDidChange = new Emitter<void>();
 	private toDispose: IDisposable[] = [];
-	// List of ignoreFiles. Used to detect changes to the ignoreFiles.
-	private ignoreFileResources: URI[] = [];
+	// List of ignoreFile resources. Used to detect changes to the ignoreFiles.
+	private ignoreFileResources: Map<String, URI> = new Map<String, URI>();
```

Comment:
ResourceSet is probably what you want here. Or at least the lowercase `string` type.